raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -488,188 +506,162 @@ def _initialize_web_rtc(self, partner_address: Address) -> None:
                 "Could not establish channel",
                 partner_address=to_checksum_address(partner_address),
             )
-            hang_up_message = {
-                "type": RTCMessageType.HANGUP.value,
-                "call_id": self.get_rtc_partner(partner_address).call_id,
-            }
-            self._send_signaling_message(partner_address, json.dumps(hang_up_message))
+            conn.send_hangup_message()
             self.close_connection(partner_address)
 
-    def _handle_closed_connection(self, partner_address: Address) -> None:
-        # XXX-UAM: Reachability check was here
-
-        # FIXME: temporary sleep to stretch two signaling processes a bit
-        #        with a unique call id for each try this wont be necessary
-        gevent.sleep(3)
-        self._maybe_initialize_web_rtc(partner_address)
-
-    def get_rtc_partner(self, partner_address: Address) -> _RTCPartner:
-        if partner_address not in self._address_to_rtc_partners:
-
-            self._address_to_rtc_partners[partner_address] = _RTCPartner(
-                partner_address,
-                self.node_address,
-                self._handle_message,
-                self._handle_candidates_callback,
-                self._handle_closed_connection,
-            )
-
-        return self._address_to_rtc_partners[partner_address]
+    def _add_connection(self, partner_address: Address, conn: _RTCConnection) -> None:
+        assert partner_address not in self._address_to_connection, "must not be there already"
```

Comment:
I might be mistaken but I think this assert could be hit, because the call in line 510 only schedules a task which will remove the connection eventually but not necessarily directly. since it is not awaited on the loop in _wrapped_initialize_web_rtc could have no context switch and try to add a new connection